arrow/src/array/array_list.rs

```diff
@@ -57,6 +60,16 @@ pub struct GenericListArray<OffsetSize> {
 }
 
 impl<OffsetSize: OffsetSizeTrait> GenericListArray<OffsetSize> {
+    /// The data type constructor of list array.
+    /// The input is the schema of the child array and
+    /// the output is the [`DataType`], List or LargeList.
+    pub const DATA_TYPE_CONSTRUCTOR: fn(Box<Field>) -> DataType = if OffsetSize::IS_LARGE
```

Comment:
```suggestion
    pub const MAKE_DATA_TYPE: fn(Box<Field>) -> DataType = if OffsetSize::IS_LARGE
```

`DATA_TYPE_CONSTRUCTOR` was somewhat hard on my eyes when reading this PR. That being said while I might prefer something like `MAKE_DATA_TYPE` it is only an opinion for your consideration